stan/math/prim/meta/base_type.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef STAN_MATH_PRIM_META_BASE_TYPE_HPP
+#define STAN_MATH_PRIM_META_BASE_TYPE_HPP
+
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/meta/is_complex.hpp>
+#include <stan/math/prim/meta/is_eigen.hpp>
+#include <stan/math/prim/meta/value_type.hpp>
+#include <stan/math/prim/meta/is_vector.hpp>
+#include <type_traits>
+#include <vector>
+
+namespace stan {
+
+/**
```

Comment:
If you want all of these to be in the type_traits group but not have to add `@ingroup` to all of them you can use

```cpp
// before code
/** \addtogroup type_trait
 *  @{
 */

// code stuff here

// after code
/** @}*/